src/V3Coverage.cpp

```diff
@@ -786,11 +786,13 @@ class CoverageVisitor final : public VNVisitor {
                 AstNodeExpr* covExprp = nullptr;
                 if (AstFuncRef* const frefp = VN_CAST(term.m_exprp, FuncRef)) {
                     AstNodeDType* const dtypep = frefp->taskp()->fvarp()->dtypep();
-                    const auto emplacedVarp = m_funcTemps.emplace(
-                        *frefp,
-                        new AstVar{fl, VVarType::XTEMP, m_exprTempNames.get(frefp), dtypep});
-                    AstVar* const varp = emplacedVarp.first->second;
-                    if (emplacedVarp.second) {
+                    const auto tempIt = m_funcTemps.find(*frefp);
+                    AstVar* varp = nullptr;
+                    if (tempIt == m_funcTemps.end()) {
+                        auto emplacedVarp = m_funcTemps.emplace(
+                            *frefp, new AstVar{fl, VVarType::MODULETEMP,
+                                               m_exprTempNames.get(frefp), dtypep});
+                        varp = emplacedVarp.first->second;
```

Comment:
FWIW normally I just emplace with a nullptr or other sentinel to avoid the double lookup. Something like:

```
const auto pair = map.emplace(key, nullptr);
if (pair.second) {
   pair.first->second = new Value{...};
}
```